### DISCUSSION TITLE
Editor wont accept input in Next JS (Both server & Client side)

### DISCUSSION THREAD

**Discussion by dukesx** [outline/rich-markdown-editor#234] [General]
When I load the editor in a NEXT JS app , the editor loads , but if i enter anything in it or even add a list or etc, an error pops up saying "Input data must be a string". Any help ? 

Version: Both 10.1.0 & 10.2.0

![Error Image](https://i.ibb.co/94NMY1Z/error.png)

**Comment by tommoor** [comment]
Unsure, you should trace the error and see what the value is where it's expecting a string. Are you passing a `defaultValue` prop?

**Reply by dukesx** [reply]
Yes i tried adding a defaulValue prop and i tried it without it as well. I tried tracing it but it seems that the error is somewhat deep inside the editor. I don't know but i did use the version 9 and it worked fine on Next JS, with no issues

**Reply by tommoor** [reply]
I think there must be something else going on, I span up a latest NextJS environment inserted an editor and didn't have any problems on server-side or statically generated pages.

Note: v9 is a completely different editor architecture, it was rebuilt from the ground up.

**Comment by dukesx** [comment]
Did you try by exporting an empty getIntitialProps/getServerSideProps function in the default export "pages" function/class ?

because one of the two functions, make the page server side. I was previously using TinyMCE 5 , i also removed it to check if there was some clash but removing it also didn't help. There is no issue with window or anything else. I am however fetching Apollo content from database inside GetServerSideProps to fetch extra data on the page i am initializing the editor on.

Note: On a page that is client rendered (without getInitialProps/getServerSideProps) , the editor works fine.

**Reply by tommoor** [reply]
> Did you try by exporting an empty getIntitialProps/getServerSideProps function in the default export "pages" function/class ?

Yep, fraid so. At this point I'll need a minimal example of the bug in next js starter repo or something to investigate.
